lib/signet/oauth_2/client.rb

```diff
@@ -181,6 +188,7 @@ def update! options = {}
         self.client_id = options[:client_id] if options.key? :client_id
         self.client_secret = options[:client_secret] if options.key? :client_secret
         self.scope = options[:scope] if options.key? :scope
+        self.target_audience = options[:target_audience] if options.key? :target_audience
```

Comment:
You may want to throw an exception when `target_audience` and `scope` are both set. They are not valid when set at the same time, and the Oauth2 endpoint (for Google, at least) responds with a 400 if you send them both in.

In PHP we throw an exception in GCE and ServiceAccount credentials classes, but not in the OAuth2 class, so that also works.

 - https://github.com/googleapis/google-auth-library-php/blob/master/src/Credentials/GCECredentials.php#L164
 - https://github.com/googleapis/google-auth-library-php/blob/master/src/Credentials/ServiceAccountCredentials.php#L126

Reply:
Thanks! Yeah, I think it's best to check these in the higher level auth library.